### PR TITLE
feat(pubsublite): Committer implementation

### PR DESCRIPTION
**Pull request by tmdiep**
Commits desired offsets based on the ack prefix offset. Commits are batched.

### CODE REVIEW

**Comment by tmdiep** [comment]
WANT_LGTM=@manuelmenzella-google 
Other reviewers optional, but feel free to leave comments.

Depends on a number of other PRs to be committed first, mainly https://github.com/googleapis/google-cloud-go/pull/3143.

Calling out potentially controversial behavior of the committer: when Stop() is called, it the stream is left open to process outstanding acks. For example, the user may call Subscriber.Stop() to stop receiving new messages from the server, and continue processing/acking all the outstanding messages. But if they never finish acking all those messages, the committer stream remains open. Perhaps we should have a timer to do a hard shutdown after N minutes? Or expose a Subscriber.Shutdown() to do the hard shutdown (but relies on the user to call this)?

**Review by manuelmenzella-google — APPROVED** [review]
Just some nits, feel free to merge. Thanks.

File: pubsublite/internal/wire/acks.go [hunk 1]
```diff
@@ -163,6 +163,13 @@ func (at *ackTracker) Release() {
 	at.outstandingAcks.Init()
 }
 
+// Done when there are no outstanding acks.
+func (at *ackTracker) Done() bool {
```
[the remark this answers is not in the document]
Comment (tmdiep) [reply pubsublite/internal/wire/acks.go]: Thanks for the review.
Ah yes, Empty is better.

File: pubsublite/internal/wire/committer.go [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	c.unsafeInitiateShutdown(serviceTerminating, nil)
+}
+
+func (c *committer) newStream(ctx context.Context) (grpc.ClientStream, error) {
+	return c.cursorClient.StreamingCommitCursor(ctx)
+}
+
+func (c *committer) initialRequest() (interface{}, bool) {
```
[the remark this answers is not in the document]
Comment (tmdiep) [reply pubsublite/internal/wire/committer.go]: I named the return values to make this clearer. The bool indicates that an initial response is required (only the partition assignment stream sets this to false).

**Pull request merged**